### PR TITLE
Dynamic sizing for compass rose based on image size

### LINKED ISSUE #3742: [Imagery] Compass rose does not resize with images
When resizing the window, the compass rose stays the same size, sometimes covering the imagery entirely.

### PR DESCRIPTION
**Pull request by charlesh88**
Fixes #3742. See issue for testing notes.

### Changes
- Compass rose now sizes and positions proportionally to the containing image, with min and max sizes;
- Refactored computed `compassDimensionsStyle` as `sizedImageDimensions` for reusability;
- Tweaked sizing of compass ordinals text and North arrow for better legibility;
- Minor tweaks to element positioning and opacity for better legibility;
- Per discussion, automated testing not added at this time due to other issues with Imagery failing automated tests;
- Smoke tested locally and on hulk/dev

### Author Checklist
1. Changes address original issue? Y
1. Unit tests included and/or updated with changes? N
1. Command line build passes? Y
1. Changes have been smoke-tested? Y

### CODE REVIEW

File: src/plugins/imagery/components/Compass/Compass.vue [hunk 1]
```diff
@@ -23,7 +23,7 @@
 <template>
 <div
     class="c-compass"
-    :style="compassDimensionsStyle"
+    :style="`width: ${ sizedImageDimensions.width }px; height: ${ sizedImageDimensions.height }px`"
```
Comment (jvigliotta) [inline src/plugins/imagery/components/Compass/Compass.vue:26]: I think this is fine for now, but if we anticipate more styles, we may want another computed "compassStyles" property that returns these formatted values, similar to the old compassDimensionsStyle() computed value. I like the rename and having it available for other purposes, so I think just adding another computed variable just for the styles would work.

File: src/plugins/imagery/components/Compass/CompassRose.vue [hunk 2]
```diff
@@ -21,130 +21,134 @@
  *****************************************************************************/
 
 <template>
-<div
-    class="c-direction-rose"
-    @click="toggleLockCompass"
+<div class="w-direction-rose"
```
Comment (jvigliotta) [inline src/plugins/imagery/components/Compass/CompassRose.vue:24]: Super duper nitpicky but can we just get the class attribute on a new line to be consistent with the other attributes in the template?

File: src/plugins/imagery/components/Compass/CompassRose.vue [hunk 3]
```diff
@@ -177,12 +185,28 @@ export default {
         }
     },
     computed: {
-        north() {
-            return this.lockCompass ? rotate(-this.cameraPan) : 0;
+        compassRoseSizing() {
```
Comment (jvigliotta) [inline src/plugins/imagery/components/Compass/CompassRose.vue:188]: Can we name this something like compassRoseSizingClasses()? I feel like the current name implies it's going to do something.

File: src/plugins/imagery/components/Compass/CompassRose.vue [hunk 4]
```diff
@@ -177,12 +185,28 @@ export default {
...
+
+            if (this.sizedImageWidth < 500) {
+                compassRoseSizing = '--rose-small';
+            }
+
+            if (this.sizedImageWidth > 1000) {
+                compassRoseSizing = '--rose-max';
+            }
+
+            return compassRoseSizing;
```
Comment (jvigliotta) [inline src/plugins/imagery/components/Compass/CompassRose.vue:202]: I remember we talked about this before, I think we can clean this up a bit. we had talked about returning right out of the if statements right? I think that might be the way to go here OR you could just chain these if's in to if else statements. that way we're not doing erroneous checks. Or a combo... something like:


``` Javascript
compassRoseSizingClasses() {
        if (this.sizedImageWidth < 300) {
            return '--rose-small --rose-min';
        } else if (this.sizedImageWidth < 500) {
            return '--rose-small';
        } else if (this.sizedImageWidth > 1000) {
            return '--rose-max';
        }
        
        return '';
}
```

**Review by jvigliotta — CHANGES_REQUESTED** [review]
This is awesome, I'm loving it. Seen it in action too! Just a couple comments here and there for code style stuff and naming stuff.

File: src/plugins/imagery/components/Compass/CompassRose.vue (hunk 2 above)
Comment (charlesh88) [reply src/plugins/imagery/components/Compass/CompassRose.vue:24]: Fixed.

File: src/plugins/imagery/components/Compass/CompassRose.vue (hunk 3 above)
Comment (charlesh88) [reply src/plugins/imagery/components/Compass/CompassRose.vue:188]: Got it, fixed.

File: src/plugins/imagery/components/Compass/CompassRose.vue (hunk 4 above)
Comment (charlesh88) [reply src/plugins/imagery/components/Compass/CompassRose.vue:202]: Ugh, looking at what I wrote now and it's terrible. Good suggestion, fixed.

**Comment by charlesh88** [comment]
@jvigliotta Have addressed all comments and pushed changes.

File: src/plugins/imagery/components/Compass/Compass.vue (hunk 1 above)
Comment (charlesh88) [reply src/plugins/imagery/components/Compass/Compass.vue:26]: No change requested here (?) so not addressing right now. If we need to control more styles on this in the future, we can address then.

**Review by jvigliotta — APPROVED** [review]

**Pull request merged**